### PR TITLE
Notify providers when added to/removed from placements

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

- Dispatch an email to Providers when they are added to/removed from a placement.

## Changes proposed in this pull request

- Added Mailers for adding and removing Providers from/to placements
- Add services to handle sending emails to every user of a provider when added to/removed from placement.
- Add condition to EditPlacementController, to action email dispatches only when the assigned provider to a placement changes.

## Guidance to review

To preview the mailer:
- Sign in as Colin (Support User)
- Navigate to Setting in the Navbar
- Click "Emails"
- Click "placement_provider_assigned_notification (opens in new tab)" to view the assigned email
- Click "placement_provider_removed_notification (opens in new tab)" to view the removed email

To trigger the email: (Notify may only work locally)
- Sign in as Anne (School User) or Colin (Support User)
- Assign an onboarded provider (with users) as a partner provider to the school
- Click on a placement
- Follow the video below:

https://github.com/user-attachments/assets/b72fb8b0-0c54-4f5e-b038-053ed4477d41

- ✉️ Emails dispatched can be viewed at.

## Link to Trello card

https://trello.com/c/VUVbq2FG/615-send-email-notifications-to-providers-when-theyre-assigned-unassigned-to-placements

## Screenshots
_Provider added email_
screencapture-placements-localhost-3000-rails-mailers-placements-user-mailer-placement-provider-assigned-notification-2024-07-26-13_38_13

_Provider removed email_
screencapture-placements-localhost-3000-rails-mailers-placements-user-mailer-placement-provider-removed-notification-2024-07-26-13_38_27

### CODE REVIEW

File: spec/support/shared_examples/system/placements/edit_a_placement_wizard.rb [hunk 1]
```diff
@@ -428,6 +437,21 @@ def when_i_select_year(year)
...
+
+  def and_the_provider_is_notified_they_have_been_removed_from_the_placement(user)
+    and_the_provider_is_notified(user, "#{school.name} has removed you from a placement")
+  end
+
+  def and_the_provider_is_notified(user, subject)
+    ActionMailer::Base.deliveries.find do |delivery|
+      delivery.to.include?(user.email) &&
+        delivery.subject == subject
+    end
```
Comment (ollietreend) [inline spec/support/shared_examples/system/placements/edit_a_placement_wizard.rb:452]: Could the [`have_enqueued_mail` matcher](https://rspec.info/features/6-1/rspec-rails/matchers/have-enqueued-mail-matcher/) be helpful here?

We use it in similar tests such as this:

https://github.com/DFE-Digital/itt-mentor-services/blob/4a8aaa15f7e8db3d39b000d4a44faa6e40a16918/spec/services/placements/partnerships/notify/create_spec.rb#L29-L44

**Review by ollietreend — COMMENTED** [review]
Looks good! Neither of these are blockers – so I'd be happy to approve if you're not inclined to change anything here.

File: spec/support/shared_examples/system/placements/edit_a_placement_wizard.rb (hunk 1 above)
Comment (JamieCleare2525) [reply spec/support/shared_examples/system/placements/edit_a_placement_wizard.rb:452]: I don't believe you can do that in a system rspec test.

**Review by ollietreend — APPROVED** [review]
Nice 😎

**Pull request merged**